ompi/mca/pml/ucx/pml_ucx_request.h

```diff
@@ -43,10 +43,10 @@ enum {
 #define PML_UCX_TAG_MASK                       0x7fffff0000000000ul
 
 
-#define PML_UCX_MAKE_SEND_TAG(_tag, _comm) \
+#define PML_UCX_MAKE_SEND_TAG(_tag, _comm, _c_index) \
     ((((uint64_t) (_tag)            ) << (PML_UCX_RANK_BITS + PML_UCX_CONTEXT_BITS)) | \
      (((uint64_t)(_comm)->c_my_rank ) << PML_UCX_CONTEXT_BITS) | \
-     ((uint64_t)(_comm)->c_index))
+     ((uint64_t)(_c_index)))
```

Comment:
since we only have 20bits here, shall we check that `cid` is always lower than 2^20? or somehow we are guaranteed that cid is within limit?

Reply:
who has only 20 bits?

Reply:
`PML_UCX_CONTEXT_BITS` is 20 bits so if `_c_index` was to ever be bigger, it would spill on the `->c_my_rank` field, maybe that's no the case?

Reply:
guess i'm curious - OMPI's communicator index has always been 32-bits. So have you just been lucky because the index never grew big enough to violate your 20-bit threshold? If so, then it seems you have a design problem as there is no guarantee that the index won't be greater than 20-bits.

Reply:
yes - pml_max_contextid

Reply:
but in connect/accept and comm_spawn operations, the context id is being set by PMIx, which provides a 32-bit ID. There are other places in the code that also set the ID, not just in the PML. So how do you handle ID's beyond 20-bit? Or is UCX simply limited to 20-bit ID's and fails if OMPI sets anything above that?

Reply:
actually the cid in the dpm.c code is being generated in the call to ```ompi_comm_nextcid```  and  using the "OLD" algorithm for allocating cids in the ```ompi_comm_nextcid_nb``` function since the MPI functions supported  in dpm.c for communicator construction always require an input communicator.